CTCCriterion.lua

```diff
@@ -0,0 +1,95 @@
+require 'warp_ctc'
```

Comment:
Do this inside the constructor so that the whole repo doesnt need to depend on warp_ctc and it is lazy initialized as needed